src/components/Inventory/withdraw/flow/SelectPurposeForm.tsx

```diff
@@ -90,7 +90,7 @@ export default function SelectPurposeForm(props: SelectPurposeFormProps): JSX.El
   };
 
   const fetchPlantingSites = useCallback(async () => {
-    if (plantingSites.length) {
+    if (plantingSites) {
```

Comment:
this was being called infinitely because we update plantingsites object.
updated the type to be PlantingSite[] or undefined